packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -187,3 +241,15 @@ class ${class_name}:
 
         for response in self._get_stub().Measure(request):
             yield self._deserialize_response(response)
+
+    def register_pin_map(self, pin_map_path: pathlib.Path) -> None:
+        """Registers the pin map with the pin map service.
+
+        Args:
+            pin_map_path: Absolute path of the pin map file.
+        """
+        pin_map_id = self._get_pin_map_client().update_pin_map(pin_map_path)
+        if self._pin_map_context is None:
+            self._pin_map_context = PinMapContext(pin_map_id=pin_map_id, sites=[0])
+        else:
+            self._pin_map_context._replace(pin_map_id=pin_map_id)
```

Comment:
This does not work because named tuples are immutable and _replace has no side effects.

Please add test coverage for this case before fixing this.

Reply:
Thanks for pointing out this issue! To fix this issue, I have assigned the replaced value back to `self._pin_map_context`. I have also added test cases to check this behavior.